test/tests/functional/pbs_reservations.py

```diff
@@ -914,46 +916,60 @@ def test_advance_resv_with_multinode_job_array(self):
         self.server.expect(JOB, {'job_state=Q': 3}, count=True,
                            extend='t', id=jid)
-        self.server.expect(JOB, {'job_state': 'Q'}, id=subjid[3])
-        self.server.delete(subjid[3])
-        self.server.expect(JOB, {'job_state': 'X'}, id=subjid[3])
+        # Suspend first sub-job
         self.server.sigjob(jobid=subjid[0], signal="suspend")
         self.server.expect(JOB, {'job_state': 'S'}, id=subjid[0])
+        # Verify that the third sub-job started running
         self.server.expect(JOB, {'job_state': 'R'}, id=subjid[2])
+
+        # Submit job array with placement type scatter in resv queue
         attrs = {ATTR_q: resv_queue, ATTR_J: '1-5',
                  'Resource_List.place': 'scatter',
                  'Resource_List.select': '2:ncpus=1'}
         j1 = Job(PBSROOT_USER, attrs)
         j1.set_sleep_time(60)
         jid2 = self.server.submit(j1)
         subjid2 = []
-        for i in range(1, 5):
+        for i in range(1, 6):
             subjid2.append(j.create_subjob_id(jid2, i))
         self.server.expect(JOB, {'job_state': 'Q'}, id=jid2)
 
+        # Resume first sub-job from first job array
         self.server.sigjob(subjid[0], 'resume')
-        self.server.expect(JOB, {'job_state': 'F'}, id=jid,
-                           extend='x', offset=120)
+        # Wait for the first job-array to finish
+        self.server.expect(JOB, {'job_state': 'F', 'exit_status': '0'},
+                           id=jid, extend='x', offset=180)
+
         self.server.expect(JOB, {'job_state': 'B'}, id=jid2)
         self.server.expect(JOB, {'job_state=R': 2}, count=True,
                            extend='t', id=jid2)
         self.server.expect(JOB, {'job_state=Q': 3}, count=True,
                            extend='t', id=jid2)
+        # Suspend first and second sub-jobs
         self.server.sigjob(jobid=subjid2[0], signal="suspend")
         self.server.expect(JOB, {'job_state': 'S'}, id=subjid2[0])
         self.server.sigjob(jobid=subjid2[1], signal="suspend")
         self.server.expect(JOB, {'job_state': 'S'}, id=subjid2[1])
-        self.server.delete(subjid2[1])
-        self.server.expect(JOB, {'job_state': 'X'}, id=subjid2[1])
+        # Verify third and fourth subjobs are running
         self.server.expect(JOB, {'job_state': 'R'}, id=subjid2[2])
+        self.server.expect(JOB, {'job_state': 'R'}, id=subjid2[3])
+        # Delete running jobs
+        self.server.delete(subjid2[2], subjid2[3])
+        # Verify fifth subjob is in R state
+        self.server.expect(JOB, {'job_state': 'R'}, id=subjid2[4])
+        # Resume the suspended jobs
         self.server.sigjob(subjid2[0], 'resume')
-        self.server.expect(JOB, {'job_state': 'F'}, id=jid2,
-                           extend='x', offset=120)
+        self.server.sigjob(subjid2[1], 'resume')
+        self.server.expect(JOB, {'job_state=R': 2}, count=True,
+                           extend='t', id=jid2)
```

Comment:
Resuming subjobs 1 and 2 will not resume them immediately.  It will change their substate from 43 to 45.  A job in substate 45 is looked at by the scheduler to resume it when there is space to do so.  If I understand properly,  there should be 3 running jobs and 2 suspended jobs right now.  What this does is wait for the 3 running jobs to end, and then the 2 suspended jobs resume.  Is this what you want?